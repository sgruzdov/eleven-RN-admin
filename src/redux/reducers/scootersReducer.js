import produce from 'immer'
import * as axios from 'axios'

import { URL } from '../../server/config'

export const IS_LOAD_SCOOTERS = 'IS_LOAD_SCOOTERS'
export const SET_SCOOTERS = 'SET_SCOOTERS'
export const REMOVE_SCOOTERS = 'REMOVE_SCOOTERS'
export const SET_SELECTED_SCOOTER = 'SET_SELECTED_SCOOTER'
export const CHANGE_STATUS_ACTIVE = 'CHANGE_STATUS_ACTIVE'

const initialState = {
    data: [],
    selectedScooter: null,
    loading: false
}


export const scootersReducer = (state = initialState, action) => {
    switch(action.type) {
        case IS_LOAD_SCOOTERS:
            return produce(state, draft => {
                draft.loading = action.payload
            })
        case SET_SCOOTERS:
            return produce(state, draft => {
                draft.data = action.payload
            })
        case REMOVE_SCOOTERS:
            return produce(state, draft => {
                draft.data = []
                draft.selectedScooter = null
            })
        case SET_SELECTED_SCOOTER:
            return produce(state, draft => {
                draft.selectedScooter = action.payload
            })
        case CHANGE_STATUS_ACTIVE:
            return produce(state, draft => {
                // draft.selectedScooter.
            })
        default:
            return state
    }
}

export const getScootersThunk = (data = {}) => async dispatch => {
    dispatch({ type: IS_LOAD_SCOOTERS, payload: true })

    try {
        const res = await axios.get(`${URL}/scooters/getScooters`, {params: JSON.stringify(data)})
        dispatch({ type: SET_SCOOTERS, payload: res.data })
    } catch (e) {
        console.log('getScootersThunk', e)
    }

    dispatch({ type: IS_LOAD_SCOOTERS, payload: false })
}

export const changeStatusActiveThunk = (id, status, typeActive) => async dispatch => {
    dispatch({ type: IS_LOAD_SCOOTERS, payload: true })
    try {
        const res = await axios.post(`${URL}/scooters/changeStatusActive`, {id, status})
        
        await dispatch(getScootersThunk(typeActive))

        dispatch({ type: SET_SELECTED_SCOOTER, payload: res.data})

    } catch (e) {
        console.log('changeStatusActiveThunk', e)
    }
    dispatch({ type: IS_LOAD_SCOOTERS, payload: false })
}