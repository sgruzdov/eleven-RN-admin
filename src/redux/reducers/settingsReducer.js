import produce from 'immer'
// import * as axios from 'axios'

export const IS_MENU = 'IS_MENU'


const initialState = {
    loading: false,
    isMenu: true,
}


export const settingsReducer = (state = initialState, action) => {
    switch(action.type) {
        case IS_MENU:
            return produce(state, draft => {
                draft.isMenu = !state.isMenu
            })
        default:
            return state
    }
}