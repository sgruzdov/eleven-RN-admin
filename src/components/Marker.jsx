import React from 'react'
import { useDispatch } from 'react-redux'

import markerGreen from '../assets/icons/marker-green.png'
import markerOrange from '../assets/icons/marker-orange.png'
import markerRed from '../assets/icons/marker-red.png'
import { SET_SELECTED_SCOOTER } from '../redux/reducers/scootersReducer'


const Marker = ({ marker }) => {
    const dispatch = useDispatch()

    const charge = () => {
        let charge = null

        if(marker.charge >= 50) {
            charge = markerGreen
        } else if(marker.charge > 29 && marker.charge < 49 ) {
            charge = markerOrange
        } else {
            charge = markerRed
        }
        return charge
    }

    return(
        <div style={{cursor: 'pointer'}} onClick={() => dispatch({ type: SET_SELECTED_SCOOTER, payload: marker })}>
            <img style={{width: '30px'}} src={charge()} alt="marker" />
        </div>
    )
}

export default Marker
