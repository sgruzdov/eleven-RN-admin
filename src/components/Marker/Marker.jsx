import React from 'react'
import { useDispatch } from 'react-redux'

import markerGreen from '../../assets/icons/marker-green.png'
import markerOrange from '../../assets/icons/marker-orange.png'
import markerRed from '../../assets/icons/marker-red.png'
import { SET_SELECTED_SCOOTER } from '../../redux/reducers/scootersReducer'
import PropTypes from 'prop-types'

const Marker = ({ marker }) => {
    const dispatch = useDispatch()

    const charge = () => {
        if (marker.charge >= 50) {
            return markerGreen
        } else if (marker.charge > 29 && marker.charge < 49) {
            return markerOrange
        } else {
            return markerRed
        }
    }

    return (
        <div
            className="marker"
            style={{ cursor: 'pointer' }}
            onClick={() => dispatch({ type: SET_SELECTED_SCOOTER, payload: marker })}
        >
            <img style={{ width: '30px' }} src={charge()} alt="marker" />
        </div>
    )
}

Marker.propTypes = {
    scooter: PropTypes.shape({
        location: PropTypes.objectOf(PropTypes.number),
        _id: PropTypes.string,
        scooterId: PropTypes.number,
        charge: PropTypes.number,
        active: PropTypes.bool,
        userActive: PropTypes.bool,
        breakdown: PropTypes.bool,
        __v: PropTypes.number,
    }),
}

export default Marker
