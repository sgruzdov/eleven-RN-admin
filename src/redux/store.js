import { createStore, combineReducers, applyMiddleware } from "redux"
import { composeWithDevTools } from "redux-devtools-extension"
import thunk from 'redux-thunk'

import { settingsReducer } from './reducers/settingsReducer'
import { scootersReducer } from './reducers/scootersReducer'

const reducer = combineReducers({
    settings: settingsReducer,
    scooters: scootersReducer,
})

export const store = createStore(reducer, composeWithDevTools(applyMiddleware(thunk)))