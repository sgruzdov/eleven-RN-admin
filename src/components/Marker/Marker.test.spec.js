import * as redux from 'react-redux'

import { store } from '../../redux/store'
import Marker from './Marker'
import { SET_SELECTED_SCOOTER } from '../../redux/reducers/scootersReducer'

const scooter = {
    location: {
        latitude: 53.853253464874754,
        longitude: 27.601949922194905,
    },
    _id: '60f5b4bad21fa65a98c3f9f5',
    scooterId: 955227,
    charge: 60,
    active: true,
    userActive: true,
    breakdown: true,
    __v: 0,
}

const mountComponent = () =>
    mount(
        <redux.Provider store={store}>
            <Marker marker={scooter} />
        </redux.Provider>
    )

describe('Marker component', () => {
    let component

    beforeEach(() => {
        component = mountComponent({})
    })

    it('should render Marker component', () => {
        expect(component).toMatchSnapshot()
    })

    describe('marker charge', () => {
        it('red', () => {
            scooter.charge = 10
            component = mountComponent()
        })

        it('orange', () => {
            scooter.charge = 30
            component = mountComponent()
        })

        it('green', () => {
            scooter.charge = 60
            component = mountComponent()
        })
    })
})
