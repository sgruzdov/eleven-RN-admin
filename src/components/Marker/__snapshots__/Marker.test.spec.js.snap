// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker component should render Marker component 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Marker
    marker={
      Object {
        "__v": 0,
        "_id": "60f5b4bad21fa65a98c3f9f5",
        "active": true,
        "breakdown": true,
        "charge": 60,
        "location": Object {
          "latitude": 53.853253464874754,
          "longitude": 27.601949922194905,
        },
        "scooterId": 955227,
        "userActive": true,
      }
    }
  >
    <div
      className="marker"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <img
        alt="marker"
        src="marker-green.png"
        style={
          Object {
            "width": "30px",
          }
        }
      />
    </div>
  </Marker>
</Provider>
`;
